### PR TITLE
[AMORO-2059]Remove Flink 1.12 related modules

### LINKED ISSUE #2059: [Subtask]: Remove Flink 1.12 modules
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

Remove Flink 1.12 related modules.

### Parent issue

#1983

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

Close #2059.

## Brief change log

- Remove codes of Flink 1.12 relasted modules
- Change module configuration in flink/pom.xml

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/engines/flink/flink-get-started.md [hunk 1]
```diff
@@ -49,7 +48,7 @@ Download Flink and related dependencies, and download Flink 1.12/1.14/1.15 as ne
```
Comment (YesOrNo828) [inline docs/engines/flink/flink-get-started.md:48]:
```suggestion
Download Flink and related dependencies, and download Flink 1.14 and 1.15 as needed. Taking Flink 1.15 as an example:
```

**Review by YesOrNo828 — COMMENTED** [review]
@zhoujinsong  I left a comment. Please take a look.

**Review by YesOrNo828 — APPROVED** [review]
LGTM.
